### PR TITLE
Wp3 storm surge

### PR DESCRIPTION
**Pull request by mathvansoest**
Merge with main before creating pull request, no issues show up now.

### CODE REVIEW

**Comment by EtienneKras** [comment]
So, with this PR you tend to merge in two datasets: TWL & storm surge and wave climate.
Couple of comments:
- could you check indentation? I see that the catalog.json file changes again nearly as a whole
- also other files like cbca, ceed, etc. changed due to, I suspect, indentation. I only expect the newly added datasets collections to change and not the other collections. Could you check this as well?
- As these sets are part of full-track WP3, could you add the following keywords to both collections for twl & ss_wc: "Sea Levels", "Full-Track"? This allows us to filter more efficiently
- in ss_wc there is a validation error from pystac in the license. Please check if you can resolve this. 

Cheers

File: src/coclicodata/etl/cloud_utils.py [hunk 1]
```diff
@@ -217,15 +234,39 @@ def load_google_credentials(google_token_fp: Union[pathlib.Path, None] = None) -
...
+    #     google_token_fp=coclico_data_dir.joinpath("google_credentials.json")
+    # )
 
-    # TODO: safe cloud creds in password client
-    load_env_variables(env_var_keys=["MAPBOX_ACCESS_TOKEN"])
-    load_google_credentials(
-        google_token_fp=coclico_data_dir.joinpath("google_credentials.json")
+    # hard-coded input params
+    GCS_PROJECT = "coclico-11207608-002"
+    BUCKET_NAME = "coclico-data-public"
```
Comment (EtienneKras) [inline src/coclicodata/etl/cloud_utils.py:250]: Could you clean this up a bit, I see double coded variables

File: scripts/utils/upload_and_generate_geojson.py [hunk 2]
```diff
@@ -105,22 +94,22 @@
...
+    #             ds = ds.sel({"n" + k: k == v})
+    #         except:
+    #             raise ValueError(f"Cannot find {k}")
+
+    # dimvals = get_dimension_values(ds, dimensions_to_ignore=DIMENSIONS_TO_IGNORE)
+    # dimcombs = get_dimension_dot_product(dimvals)
 
     for var in VARIABLES:
         collection = get_geojson(
-            ds, variable=var, dimension_combinations=dimcombs, stations_dim="stations"
```
Comment (EtienneKras) [inline scripts/utils/upload_and_generate_geojson.py:123]: Why did you change stations to locs? Is this not stations in the CF compliant file?

File: scripts/utils/upload_and_generate_geojson.py [hunk 3]
```diff
@@ -105,22 +94,22 @@
...
-                raise ValueError(f"Cannot find {k}")
-
-    dimvals = get_dimension_values(ds, dimensions_to_ignore=DIMENSIONS_TO_IGNORE)
-    dimcombs = get_dimension_dot_product(dimvals)
+    # for k, v in MAP_SELECTION_DIMS.items():
+    #     if k in ds.dims and ds.coords:
+    #         ds = ds.sel({k: v})
+    #     else:
+    #         try:
+    #             # assume that coordinates with strings always have same dim name but with n
```
Comment (EtienneKras) [inline scripts/utils/upload_and_generate_geojson.py:102]: Why is this commented out? In principle it should go over this correctly

File: current/twl/collection.json [hunk 4]
```diff
@@ -0,0 +1,632 @@
+{
```
Comment (EtienneKras) [inline current/twl/collection.json:1]: I cannot find the keywords, did this go well?

File: current/ss_wc/collection.json [hunk 5]
```diff
@@ -0,0 +1,632 @@
+{
```
Comment (EtienneKras) [inline current/ss_wc/collection.json:1]: I cannot find the keywords, did this go well?

**Review by EtienneKras — CHANGES_REQUESTED** [review]
Nice work, nearly there I think

File: scripts/utils/upload_and_generate_geojson.py (hunk 2 above)
Comment (mathvansoest) [reply scripts/utils/upload_and_generate_geojson.py:123]: No in the CF-compliant file, it is named: locs.

File: scripts/utils/upload_and_generate_geojson.py (hunk 3 above)
Comment (mathvansoest) [reply scripts/utils/upload_and_generate_geojson.py:102]: Dataset only has one single dimension, no need to select certain ones or make combinations.

File: src/coclicodata/etl/cloud_utils.py (hunk 1 above)
Comment (mathvansoest) [reply src/coclicodata/etl/cloud_utils.py:250]: Check!

File: current/twl/collection.json (hunk 4 above)
Comment (mathvansoest) [reply current/twl/collection.json:1]: Check!

File: current/ss_wc/collection.json (hunk 5 above)
Comment (mathvansoest) [reply current/ss_wc/collection.json:1]: Check!

File: scripts/utils/upload_and_generate_geojson.py (hunk 2 above)
Comment (EtienneKras) [reply scripts/utils/upload_and_generate_geojson.py:123]: I do think we agreed on naming it stations to allow the F/E to work without any hard-coded if/else statements.. Can you revise the CF compliant file then?

File: scripts/utils/upload_and_generate_geojson.py (hunk 3 above)
Comment (EtienneKras) [reply scripts/utils/upload_and_generate_geojson.py:102]: If the parameters are filled in correcly, there is no need to comment this out. It should pass over without any hassle. Not sure if its correctly formatted now.. Lets have a call about it.

File: notebooks/B01_epsg=3035_x=2366962_y=2138862.tif
Comment (EtienneKras) [inline notebooks/B01_epsg=3035_x=2366962_y=2138862.tif:1]: What is this file doing here?

File: current/ss_wc/SSp99-mapbox/SSp99-mapbox.json [hunk 6]
```diff
@@ -0,0 +1,112 @@
+{
```
Comment (EtienneKras) [inline current/ss_wc/SSp99-mapbox/SSp99-mapbox.json:1]: I would expect the files to be named SSp99-mapbox-value.json. Like in EESL where the combination of variables (in this case just plain "value") is referred to..

File: current/twl/RP1-mapbox/RP1-mapbox.json [hunk 7]
```diff
@@ -0,0 +1,112 @@
+{
```
Comment (EtienneKras) [inline current/twl/RP1-mapbox/RP1-mapbox.json:1]: Why dont we have RP1, 100 and 1000 as variables for this dataset? Such that we can select them as dropdown in a viewer? Then the mapbox files with different RP would be three in the folder twl-mapbox (which is the variable). I think this has to do with the CF compliant file?

**Review by EtienneKras — COMMENTED** [review]
Two more comments on the dataset formatting

File: current/twl/RP1-mapbox/RP1-mapbox.json (hunk 7 above)
Comment (EtienneKras) [reply current/twl/RP1-mapbox/RP1-mapbox.json]: Nicely fixed

**Comment by EtienneKras** [comment]
Final changes in the PY script for SS_WC required

**Review by EtienneKras — COMMENTED** [review]
Final changes in the PY files

**Review by EtienneKras — APPROVED** [review]

**Comment by EtienneKras** [comment]
Adding SS_WC & TWL datasets
